tensorflow/compiler/mlir/tosa/transforms/legalize_common.cc

```diff
@@ -2494,12 +2494,13 @@ llvm::Optional<Value> convertFloorModOp(PatternRewriter& rewriter,
                                         Operation* op, Value result_value,
                                         Value lhs_value, Value rhs_value) {
   // FloorMod lowering:
-  // (1/rhs * lhs) - floor(1/rhs * lhs)
+  // ((1/rhs * lhs) - floor(1/rhs * lhs)) * rhs
   // a1 = reciprocal(rhs);
   // a2 = mul(lhs, a1);
```

Comment:
Any reason a2 has the multiplication order inverted from the formula in 2497?

Reply:
In the existing implementation floormod equation is` floormod(x,y) = (x/y) - floor(x/y)` which is wrong. 
According to the actual formula this PR would equate by multiplying rhs (y) to the whole equation as below.

`floormod(x,y) = ((x/y) - floor(x/y))*y`   -->  `floormod(x,y) = x- floor(x/y)*y`